cmd/trace-agent/test/runner.go

```diff
@@ -179,6 +179,9 @@ func (s *Runner) DoReq(url, method string, payload []byte) (*http.Response, erro
 
 func (s *Runner) doRequest(req *http.Request) error {
```

Comment:
I wonder if this code should look like this instead:

```
func (s *Runner) doRequest(req *http.Request) error {
	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		return err
    }
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusOK {
		slurp, err := ioutil.ReadAll(resp.Body)
		if err != nil {
			return fmt.Errorf("%s (error reading response body: %v)", resp.Status, err)
		}
		return fmt.Errorf("%s: %s", resp.Status, slurp)
	}
	return nil
}
```

The [docs](https://pkg.go.dev/net/http#Client.Do) say "On error, any Response can be ignored" and yet we are trying to read the response body and status code without first checking that `Do` returned a nil error. That seems like a separate bug here. I'm happy to change that in a separate PR if you don't want to mess with that here though.

Reply:
Created https://github.com/DataDog/datadog-agent/pull/12850